src/main/java/anichan/Main.java

```diff
@@ -123,17 +125,21 @@ private void loadUserData() {
         ArrayList<Workspace> workspaceList = new ArrayList<>();
         String[] workspaceNameList = storageManager.retrieveWorkspaceList();
         for (String workspaceName : workspaceNameList) {
-            ui.printMessage("Workspace \"" + workspaceName + "\":");
+            boolean workspaceValid = workspaceName.matches(REGEX_ALPHANUMERIC_WITH_SPACE);
+            if (workspaceValid && workspaceName.length() <= MAXIMUM_WORKSPACE_NAME_LENGTH) {
+                ui.printMessage("Workspace \"" + workspaceName + "\":");
 
-            ArrayList<Watchlist> watchlistList = loadWatchlistData(workspaceName);
-            Bookmark bookmark = loadBookmarkData(workspaceName);
+                ArrayList<Watchlist> watchlistList = loadWatchlistData(workspaceName);
+                Bookmark bookmark = loadBookmarkData(workspaceName);
 
-            if (watchlistList.size() == 0) {
-                watchlistList.add(new Watchlist("Default"));
+                if (watchlistList.size() == 0) {
+                    watchlistList.add(new Watchlist("Default"));
+                }
+                Workspace workspace = new Workspace(workspaceName, watchlistList, bookmark);
+                workspaceList.add(workspace);
             }
```

Comment:
Great work on adding more checks but I think you might want to consider using a guard clause here instead of indenting the whole statement? Because this can become a arrowhead code if more checks are added in the future.

Reply:
I've moved into StorageManager to avoid cluttering Main(). Is that better now?